include/nuttx/fs/fs.h

```diff
@@ -462,6 +462,7 @@ struct file
   int               f_oflags;   /* Open mode flags */
 #ifdef CONFIG_FS_REFCOUNT
   int               f_refs;     /* Reference count */
+  spinlock_t        lock;       /* Reference spinlock */
```

Comment:
can we sue atomic_t instead spinlock?

Reply:
https://github.com/apache/nuttx/pull/15158 
I tried to make some modifications, but i didn't achieve the intended purpose.

Reply:
could try atomic_cmpxchg, @hujun260

Reply:
done